pathfinding_service.rst

```diff
@@ -219,6 +223,47 @@ Example
         "errors": "No suitable path found."
     }
 
+
+``GET api/v1/<token_network_address>/payment/iou``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Request the last IOU used by ``sender`` to pay the PFS.
+This IOU can be used by the client to generate the next IOU to pay the PFS by increasing the ``amount`` and updating the signature.
+
+Arguments
+"""""""""
+
++---------------------+------------+---------------------------------------------------------+
+| Field Name          | Field Type | Description                                             |
++=====================+============+=========================================================+
+| sender              | address    | Sender of the payment (Ethereum address of client)      |
++---------------------+------------+---------------------------------------------------------+
+| receiver            | address    | Receiver of the payment (Ethereum address of PFS)       |
++---------------------+------------+---------------------------------------------------------+
+| timestamp           | string     | Current UTC date and time in ISO 8601 format            |
```

Comment:
Why is the timestamp needed?

Reply:
Without it anyone who somehow got hold of a request could replay it indefinitely to get information about IOUs for this user. This way, the PFS can check that the request is recent and refuse old requests.

Reply:
Ok, makes sense.

Reply:
Should we propose a maximum time delta after which the request should be ignored. Something like 10s?

Reply:
Instructing the client to send the current time is enough for the spec IMO. If we want to specify something, I suggest to be less strict (e.g. 10min) since this only guards against long term information leaks. I don't want to prevent clients from working only because their local clock is off by a minute.